gradio/blocks.py

```diff
@@ -2162,7 +2162,10 @@ def launch(
         share: bool | None = None,
         debug: bool = False,
         max_threads: int = 40,
-        auth: Callable | tuple[str, str] | list[tuple[str, str]] | None = None,
+        auth: Callable[..., Any]
```

Comment:
We can be more specific here:
```suggestion
        auth: Callable[[str, str], bool]
```